pkg/operator/resource/resourceapply/monitoring.go

```diff
@@ -0,0 +1,188 @@
+package resourceapply
+
+import (
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+var serviceMonitorGVR = schema.GroupVersionResource{Group: "monitoring.coreos.com", Version: "v1", Resource: "servicemonitors"}
+
+func mergeUnstructuredSlice(required, existing []interface{}) ([]interface{}, bool, error) {
```

Comment:
@sanchezl can we just use mergo?

@sanchezl did you pin mergo in glide.yaml to the level used in kube?  if not, please do so.

Reply:
mergo worked for me. The use of mergo got [moved into library-go](https://github.com/openshift/library-go/blob/ed90b992b08815b36c2b71c80218ea2135d8e4bd/pkg/operator/configobserver/config_observer_controller.go#L103).